### DISCUSSION TITLE
Welcome to crc Discussions!

### DISCUSSION THREAD

**Discussion by Nicoretti** [Nicoretti/crc#39] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by bgllkuka** [comment]
Hey to Nicoretti and the community,
i've some questions about the parameters in the custom config area. What is the init_value means and how is it implemented? what's the function of the reverse_input and the reverse_output parameter? Could anybody explain me the difference between crcmod implementation and this implementation?

**Reply by Nicoretti** [reply]
Hi @bgllkuka,

welcome! Generally I would recommend creating a question rather than putting your question as a comment to an announcement. This will help other people to find and respond to it more easily, also it will keep the discussions more helpful to everybody.  That said, let's switch to your questions:

>  What is the init_value means and how is it implemented?

It is the value the `crc-register` is initialized with. Depending on it's value it will impact the outcome of the crc calculation.
You can think of it similarly as an init value for "counting", depending where you start to count `0` or `5` the result of counting 
5 times will be either `5` or `10`. Regarding the implementation you can have a look [here](https://github.com/Nicoretti/crc/blob/795f3f36c4088cb4bd45692d5d57b8f158737df6/crc.py#L207).

>  what's the function of the reverse_input and the reverse_output parameter?

The reverse output is yet another setting, changing the behavior of the crc algorithm, depending what one is
optimizing for reversing the output may make sense.

You will find the implementation [here](https://github.com/Nicoretti/crc/blob/795f3f36c4088cb4bd45692d5d57b8f158737df6/crc.py#L244).

> Could anybody explain me the difference between crcmod implementation and this implementation?

I am not familiar with the crcmod implementation, at first glance major differences look like:

* It can generate `C/C++` code for crc algorithms
* It works with `python2` and `python3`


## Note 
The specific configuration settings of CRC algorithms are rather interesting for selecting and/or "optimizing"  how the CRC will "behave". As long as you are unsure about those settings I would recommend to use pre defined settings like `CCITT`.
If you wan't/need to tweak your CRC configuration settings, I would recommend to read up on the [CRC](https://www.zlib.net/crc_v3.txt) topic.
